sdcm/cluster_k8s/__init__.py

```diff
@@ -414,8 +414,27 @@ def deploy_scylla_manager(self, pool_name: str = None) -> None:
                 pool_name = self.AUXILIARY_POOL_NAME
             LOGGER.info("Deploy scylla-manager")
 
-            helm_affinity = get_helm_pool_affinity_values(self.POOL_LABEL_NAME, pool_name) if pool_name else {}
+            helm_affinity = get_helm_pool_affinity_values(
+                self.POOL_LABEL_NAME, pool_name) if pool_name else {}
             values = HelmValues(**helm_affinity)
+            if helm_affinity.get("affinity"):
```

Comment:
only way affinity is empty is pool_name is empty
So, having the following we be much clearer:
```
values = HelmValues({})
if pool_name:
    values = HelmValues(get_helm_pool_affinity_values(self.POOL_LABEL_NAME, pool_name))
    values.set("controllerAffinity", helm_affinity["affinity"])
    ....
```

Reply:
Removed condition at all because "pool_name" is set above